### PR TITLE
style(scripts): Align gen-upgrade-proposal.sh with community forum release posts

### LINKED ISSUE #9270: gen-upgrade-proposal.sh variable name spelling is inconsistent with forum posts
## What is the Problem Being Solved?

Compare [scripts/gen-upgrade-proposal.sh](https://github.com/Agoric/agoric-sdk/blob/947973cdb72b040410cd56de221c2b154582dff8/scripts/gen-upgrade-proposal.sh) to e.g. [[Proposal #71] agoric-upgrade-14](https://community.agoric.com/t/proposal-71-agoric-upgrade-14-interchain-stack-smart-wallet-upgrades/633#creating-an-emerynet-upgrade-proposal-for-rc1-3)
```diff
 COMMIT_ID=$(git rev-parse HEAD)
-ZIPURL=https://github.com/Agoric/agoric-sdk/archive/$COMMIT_ID.zip
 
-echo "Verifying archive is at $ZIPURL..." 1>&2
-zipfile=$(mktemp)
-trap 'rm -f "$zipfile"' EXIT
-curl -L "$ZIPURL" -o "$zipfile"
+ZIP_URL="https://github.com/Agoric/agoric-sdk/archive/${COMMIT_ID}.zip"
 
-echo "Generating SHA-256 checksum..." 1>&2
-checksum=sha256:$(shasum -a 256 "$zipfile" | cut -d' ' -f1)
+CHECKSUM=sha256:$(curl -L "$ZIP_URL" -o- | shasum -a 256 | cut -d' ' -f1)
 
-info="{\"binaries\":{\"any\":\"$ZIPURL//agoric-sdk-$COMMIT_ID?checksum=$checksum\"},\"source\":\"$ZIPURL?checksum=$checksum\"}"
+UPGRADE_INFO="{\"binaries\":{\"any\":\"$ZIP_URL//agoric-sdk-${COMMIT_ID}?checksum=$CHECKSUM\"},\"source\":\"$ZIP_URL?checksum=$CHECKSUM\"}"
+top-node-on-err = {{ .Streaming.ABCI.StopNodeOnErr }}
```

## Description of the Design

Since we're demonstrably comfortable with sharing the forum post representation, adopt its patterns into gen-upgrade-proposal.sh. 

## Security Considerations

None.

## Scaling Considerations

n/a

## Test Plan

This code is well-exercised.

## Upgrade Considerations

Functionality must not change.

### PR DESCRIPTION
**Pull request by gibson042**
Fixes #9270

## Description

Adopt variable names and interpolation patterns as seen in community forum posts such as cf. https://community.agoric.com/t/proposal-71-agoric-upgrade-14-interchain-stack-smart-wallet-upgrades/633#creating-an-emerynet-upgrade-proposal-for-rc1-3 . Also provides slightly more guidance regarding expected transaction options and where to look for past proposals.

### Security Considerations

n/a

### Scaling Considerations

n/a

### Documentation Considerations

We might want to direct operators to this script, rather than duplicating its contents.

### Testing Considerations

This script is well-exercised.

### Upgrade Considerations

Script functionality must not change.

### CODE REVIEW

File: scripts/gen-upgrade-proposal.sh [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
 trap 'rm -f "$zipfile"' EXIT
-curl -L "$ZIPURL" -o "$zipfile"
+curl -L "$ZIP_URL" -o "$zipfile"
 
 echo "Generating SHA-256 checksum..." 1>&2
-checksum=sha256:$(shasum -a 256 "$zipfile" | cut -d' ' -f1)
+CHECKSUM=sha256:$(shasum -a 256 "$zipfile" | cut -d' ' -f1)
 
-info="{\"binaries\":{\"any\":\"$ZIPURL//agoric-sdk-$COMMIT_ID?checksum=$checksum\"},\"source\":\"$ZIPURL?checksum=$checksum\"}"
+UPGRADE_INFO="{\"binaries\":{\"any\":\"$ZIP_URL//agoric-sdk-${COMMIT_ID}?checksum=$CHECKSUM\"},\"source\":\"$ZIP_URL?checksum=$CHECKSUM\"}"
```
Comment (gibson042) [inline scripts/gen-upgrade-proposal.sh]: This aligns with recent forum posts, but we could also quote all interpolated variables (both here and in future posts):
```suggestion
UPGRADE_INFO="{\"binaries\":{\"any\":\"${ZIP_URL}//agoric-sdk-${COMMIT_ID}?checksum=${CHECKSUM}\"},\"source\":\"${ZIP_URL}?checksum=${CHECKSUM}\"}"
```

File: scripts/gen-upgrade-proposal.sh [hunk 2]
```diff
@@ -1,43 +1,45 @@
...
-echo "Verifying archive is at $ZIPURL..." 1>&2
+echo "Verifying archive is at $ZIP_URL..." 1>&2
 zipfile=$(mktemp)
 trap 'rm -f "$zipfile"' EXIT
-curl -L "$ZIPURL" -o "$zipfile"
+curl -L "$ZIP_URL" -o "$zipfile"
 
 echo "Generating SHA-256 checksum..." 1>&2
-checksum=sha256:$(shasum -a 256 "$zipfile" | cut -d' ' -f1)
+CHECKSUM=sha256:$(shasum -a 256 "$zipfile" | cut -d' ' -f1)
```
Comment (gibson042) [inline scripts/gen-upgrade-proposal.sh]: The code in forum posts is not assumed to be running inside a script, and so doesn't have this download-with-deferred-cleanup behavior. But unfortunately, that means it also doesn't have verification that the archive exists. We could get that by adding a HEAD request both here and in future posts, e.g.
```sh
echo "Verifying archive is at $ZIP_URL..." 1>&2
curl -fLI --no-progress-meter "$ZIP_URL" -o- > /dev/null

echo "Generating SHA-256 checksum..." 1>&2
CHECKSUM=sha256:$(curl -fL "$ZIP_URL" -o- | shasum -a 256 | cut -d' ' -f1)
```
Or just keeping the data (currently ~10 MB) in a variable:
```sh
ZIP_DATA=$(curl -fL "$ZIP_URL" -o-)
CHECKSUM=sha256:$(printf '%s' "$ZIP_DATA" | shasum -a 256 | cut -d' ' -f1)
```

**Review by turadg — APPROVED** [review]
Looks fine to me. I pushed a white space fix to pass lint.

File: scripts/gen-upgrade-proposal.sh (hunk 2 above)
Comment (michaelfig) [reply scripts/gen-upgrade-proposal.sh]: I'd prefer using the `-f` flag to have `curl` fail fast:

```sh
echo "Verifying ${ZIP_URL} presence and generating SHA-256 checksum..." 1>&2
CHECKSUM=sha256:$(curl -fL "$ZIP_URL" -o- | shasum -a 256 | cut -d' ' -f1)
```

File: scripts/gen-upgrade-proposal.sh (hunk 1 above)
Comment (michaelfig) [reply scripts/gen-upgrade-proposal.sh]: As you wish.  It may help readability, though is semantically identical, AFAIK.

**Review by michaelfig — APPROVED** [review]

File: scripts/gen-upgrade-proposal.sh (hunk 2 above)
Comment (gibson042) [reply scripts/gen-upgrade-proposal.sh]: As noted above, the code in forum posts is not assumed to be running inside a script—so `pipefail` may not be active or even available (our instructions don't require bash). For example, here's what I see when running in a clean shell:
```console
$ CHECKSUM=sha256:$(curl -L 'https://example.invalid/' -o- | shasum -a 256 | cut -d' ' -f1)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (6) Could not resolve host: example.invalid
$ echo "CHECKSUM: $CHECKSUM"
CHECKSUM: sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```

But I do like the idea of using `-f`, and have updated accordingly.

File: scripts/gen-upgrade-proposal.sh (hunk 1 above)
Comment (gibson042) [reply scripts/gen-upgrade-proposal.sh]: It is. Such a change would only be for consistency and readability.

**Pull request merged**